tests/nexus_api_test.go

```diff
@@ -26,48 +26,187 @@ package tests
 
 import (
 	"context"
+	"encoding/json"
 	"errors"
 	"fmt"
 	"net/http"
+	"net/url"
 	"testing"
+	"time"
 
+	"github.com/google/uuid"
 	"github.com/nexus-rpc/sdk-go/nexus"
+	commonpb "go.temporal.io/api/common/v1"
+	enumspb "go.temporal.io/api/enums/v1"
+	nexuspb "go.temporal.io/api/nexus/v1"
+	taskqueuepb "go.temporal.io/api/taskqueue/v1"
+	"go.temporal.io/api/workflowservice/v1"
+	sdkclient "go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/converter"
 	"go.temporal.io/server/common/authorization"
 	"go.temporal.io/server/common/metrics"
+	"go.temporal.io/server/common/metrics/metricstest"
+	"google.golang.org/protobuf/types/known/structpb"
 )
 
 var op = nexus.NewOperationReference[string, string]("my-operation")
 
-// This test will be modified once nexus task dispatching is implemented in matching.
-func (s *clientFunctionalSuite) TestNexusStartOperation_WithNamespaceAndTaskQueue_NotImplemented() {
-	taskQueue := s.randomizeStr("task-queue")
-	url := fmt.Sprintf("http://%s/api/v1/namespaces/%s/task-queues/%s/dispatch-nexus-task", s.httpAPIAddress, s.namespace, taskQueue)
-	client, err := nexus.NewClient(nexus.ClientOptions{ServiceBaseURL: url})
+func (s *clientFunctionalSuite) mustToPayload(v any) *commonpb.Payload {
+	conv := converter.GetDefaultDataConverter()
+	payload, err := conv.ToPayload(v)
 	s.Require().NoError(err)
-	ctx := NewContext()
-	capture := s.testCluster.host.captureMetricsHandler.StartCapture()
-	defer s.testCluster.host.captureMetricsHandler.StopCapture(capture)
-	_, err = nexus.StartOperation(ctx, client, op, "input", nexus.StartOperationOptions{})
-	var unexpectedResponse *nexus.UnexpectedResponseError
-	s.Require().ErrorAs(err, &unexpectedResponse)
-	s.Require().Equal("internal server error", unexpectedResponse.Failure.Message)
+	return payload
+}
 
-	snap := capture.Snapshot()
+func (s *clientFunctionalSuite) TestNexusStartOperation_WithNamespaceAndTaskQueue_Outcomes() {
+	type testcase struct {
+		outcome   string
+		handler   func(*workflowservice.PollNexusTaskQueueResponse) (*nexuspb.Response, *nexuspb.HandlerError)
+		assertion func(*nexus.ClientStartOperationResult[string], error)
+	}
 
-	s.Equal(1, len(snap["nexus_requests"]))
-	s.Equal(map[string]string{"namespace": s.namespace, "method": "StartOperation", "outcome": "internal_error"}, snap["nexus_requests"][0].Tags)
-	s.Equal(int64(1), snap["nexus_requests"][0].Value)
-	s.Equal(metrics.MetricUnit(""), snap["nexus_requests"][0].Unit)
+	testCases := []testcase{
+		{
+			outcome: "sync_success",
+			handler: nexusEchoHandler,
+			assertion: func(res *nexus.ClientStartOperationResult[string], err error) {
+				s.Require().NoError(err)
```

Comment:
In case you don't already know, you could do the `*require.Assertions` trick that other tests do if the majority of your assertions are going to use `.Require()` to save some bytes. It still allows you to do `.Assert()` if you need that instead, but it looks like you want require by default here.

Reply:
Yeah, I had noticed that the default behavior changed to require and not assert but didn't want to rely on it. But you have a great point, I'll remove all of these `Require()` calls.